copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookSource.java

```diff
@@ -99,77 +119,195 @@ public void transform(KeyValue<LongWritable, Map<String, AbstractFieldValue>> in
-        return value.asDouble();
-      case BOOLEAN:
-        return value.asBoolean();
-      case LONG:
-        return value.asLong();
-      case FLOAT:
-        return value.asFloat();
-      case STRING:
-        return value.asString();
+  }
+
+  /**
+   * Get the field Schema.Type from the copybook data types
+   *
+   * @param type AbstractFiledValue type to be converted to CDAP Schema.Type
+   * @return CDAP Schema.Type objects
+   */
+  private Schema.Type getFieldSchemaType(int type) {
+    switch (type) {
+      case 0:
+      case 1:
+      case 2:
+      case 3:
+      case 71:
+      case 72:
+      case 73:
+      case 74:
+      case 75:
+        return Schema.Type.STRING;
+      case 5:
+      case 6:
+      case 7:
+      case 15:
+      case 16:
+      case 23:
+        return Schema.Type.INT;
+      case 8:
+      case 11:
+      case 18:
+      case 19:
+      case 20:
+      case 22:
+      case 24:
+      case 25:
+      case 26:
+      case 27:
+      case 28:
+      case 29:
+      case 31:
+      case 32:
+      case 33:
+        return Schema.Type.DOUBLE;
+      case 17:
+        return Schema.Type.FLOAT;
+      case 4:
+      case 35:
+      case 36:
+      case 39:
+        return Schema.Type.LONG;
+      case 111:
+      case 112:
+      case 114:
+        return Schema.Type.BOOLEAN;
       default:
-        throw new IOException(String.format("Unsupported schema: %s for field: \'%s\'", field.getSchema(),
-                                            field.getName()));
+        return Schema.Type.STRING;
     }
   }
 
   /**
-   * Create schema from list field names retrieved from CopybookFiles
+   * Get the field values for the fields in the required format.
+   * Date will be returned in the format - "yyyy-MM-dd".
    *
-   * @param fieldsSet set of field names extracted from Copybook
-   * @return output schema fields
+   * @param value AbstractFieldValue object to be converted in the JAVA primitive data types
+   * @return data objects supported by CDAP
+   * @throws ParseException
    */
-  private Schema getSchema(Set<String> fieldsSet) {
-    List<Schema.Field> fields = Lists.newArrayList();
-    for (String field : fieldsSet) {
-      fields.add(Schema.Field.of(field, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+  private Object getFieldValue(@Nullable AbstractFieldValue value) throws ParseException {
+    int type = value.getFieldDetail().getType();
+    SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    SimpleDateFormat simpleDateFormat;
+    switch (type) {
+      case 0:
+      case 1:
+      case 2:
+      case 3:
+      case 71:
+        return value.asString();
+      case 4:
+        return Long.parseLong(value.asHex(), 16);
+      case 5:
+      case 6:
+      case 7:
+        return value.asInt();
+      case 15:
+      case 16:
+      case 23:
+        return Integer.parseInt(Base64.decodeBase64(Base64.encodeBase64(value.toString().getBytes())).toString());
```

Comment:
Assuming the data is stored as "10" for integer value = 2, 
Tried to decode the binary data directly using below code, it returns  "�", and converting it to integer gives NumberFormatException -
	byte[] decodedBytes = Base64.decodeBase64(Integer.toBinaryString(2).getBytes());

Whereas, when tried to first encode and then decode the binary data with below code, it returned the correct integer value = 2 - 
	byte[] encodedBytes = Base64.encodeBase64(Integer.toBinaryString(2).getBytes());
	byte[] decodedBytes = Base64.decodeBase64(encodedBytes);
	Integer.parseInt(new String(decodedBytes), 2)

Hence, encoding and decoding the binary data in the same line, and parsing it to integer to get the required result.